### PR TITLE
bash-snippets 1.15.1 (new formula)

### PR DESCRIPTION
**Pull request by alexanderepstein**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by ilovezfs** [comment]
Same review comments as here: https://github.com/Homebrew/homebrew-core/pull/15107#issuecomment-312335552

**Comment by alexanderepstein** [comment]
> I'd like to see this use the upstream install script (looks like the script needs a non-interactive mode) and pass a --prefix= parameter 

Does this mean you want to use the ```install.sh``` that I have with the repository to install this?

**Comment by ilovezfs** [comment]
@alexanderepstein yes, instead of a bin.install loop and a man.install.

**Comment by alexanderepstein** [comment]
I am not too proficient in Ruby, so after a little searching for running a bash script within a ruby script I came up with a result of ```%x( ./install.sh all )``` would that be correct?

**Comment by ilovezfs** [comment]
```
system "./install-sh", "--prefix=#{prefix}", "all"
```

**Comment by alexanderepstein** [comment]
Does it look correct now? I got rid of the build without options because I am passing the all argument to the install script which installs all of the tools.

**Comment by ilovezfs** [comment]
@alexanderepstein yes. You'll need to teach the script to accept the prefix option, though :)

**Comment by alexanderepstein** [comment]
Do you mean that my install script should be able to be run as ```./install.sh --prefix=all```? If so I can add that to the script. Although right now the script can handle ```./install.sh all``` would that work if I changed this ```system "./install.sh", "--prefix=#{prefix}", "all"``` to ```system "./install.sh all"```  or ```system "./install.sh", "all"```

**Comment by alexanderepstein** [comment]
I believe I added this correctly to my install script. I then updated the version and changed the sha256 to the newer versions.

**Comment by ilovezfs** [comment]
@alexanderepstein cool, thanks! Now on to the other review comment :)

>the test in the test do block should exercise and verify the software's functionality

Take a look at the `jose` and `vim` formulae for some good examples:
https://github.com/Homebrew/homebrew-core/blob/master/Formula/jose.rb#L27-L36
https://github.com/Homebrew/homebrew-core/blob/master/Formula/vim.rb#L124-L143

**Comment by alexanderepstein** [comment]
Will look into writing these. My build is failing because the directory that the file is trying to be copied over to does not exist. How can I create the directories before running my install script in Ruby?

**Comment by ilovezfs** [comment]
@alexanderepstein your script will need to make sure to create the directories first using `mkdir -p $prefix/bin $prefix/share/man/man1`. Also, you can't use `echo` in the script [here](https://github.com/alexanderepstein/Bash-Snippets/blob/master/install.sh#L78) since that will add a trailing new line character, which is the other reason the script is failing.

You can set the shebang at the top of the script to

```
#!/usr/bin/env bash -x
```
to debug it.

**Comment by alexanderepstein** [comment]
Would ```echo -n``` work (this prevents a new line)? And would it work to add something like ```Dir.mkdir '#{prefix}/bin'``` and ```Dir.mkdir '#{prefix}/share/man/man1'```. It is easier for me to update the ruby script rather than change the installer and release a new version.

**Comment by ilovezfs** [comment]
>Would echo -n work (this prevents a new line)?

>And would it work to add something like Dir.mkdir '#{prefix}/bin' and Dir.mkdir '#{prefix}/share/man/man1'. It is easier for me to update the ruby script rather than change the installer and release a new version.

It will need to be part of the install script. (`bin.mkpath` and `man1.mkpath` would do it in the ruby but that's not OK for a new formula.)

**Comment by ilovezfs** [comment]
>Would echo -n work (this prevents a new line)?

Yes that should work.

**Comment by alexanderepstein** [comment]
Build still failed it seems like either I don't have the necessary permissions to create the directory or it isn't attaching the prefix before creating the directory

**Comment by ilovezfs** [comment]
@alexanderepstein I think this diff will fix it:
```
diff --git a/install.sh b/install.sh
index 614519f..9ad32ae 100755
--- a/install.sh
+++ b/install.sh
@@ -76,6 +76,7 @@ response=$( echo "$@" | grep -Eo "\-\-prefix")
 
 if [[ $response == "--prefix" ]];then
   prefix=$(echo -n "$@" | sed  s/--prefix=//g | sed  s/all//g  )
+  prefix=$(echo "$@" | sed -e 's/--prefix=\(.*\) .*/\1/')
   mkdir -p $prefix/bin $prefix/share/man/man1
   for tool in "${tools[@]}"
   do
```

**Comment by alexanderepstein** [comment]
Working now with your fix :smile:

**Comment by ilovezfs** [comment]
> I have 15 tools at the moment that are packaged in bash-snippets do i test all of them or only some?

Maybe three of the most popular ones?

**Comment by alexanderepstein** [comment]
That works I will look into adding these tests as soon as possible

**Comment by ilovezfs** [comment]
Thanks @alexanderepstein!

File: Formula/bash-snippets.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class BashSnippets < Formula
+  desc "Collection of small bash scripts for heavy terminal users"
+  homepage "https://github.com/alexanderepstein/Bash-Snippets"
+  url "https://github.com/alexanderepstein/Bash-Snippets/archive/v1.15.1.tar.gz"
+  sha256 "ecc771d8d8486229309bdcaef330492ce37fb827e6ce276f251380db4a63930b"
+
+  bottle :unneeded
+
```
Comment (ilovezfs) [inline Formula/bash-snippets.rb]: This extra blank line should be deleted.

Comment (alexanderepstein) [inline Formula/bash-snippets.rb]: Line 6 or Line 8 or both?

Comment (ilovezfs) [inline Formula/bash-snippets.rb]:
```
iMac-TMP:homebrew-core joe$ brew test bash-snippets
Testing bash-snippets
==> Using the sandbox
==> /usr/local/Cellar/bash-snippets/1.15.1/bin/crypt -h
iMac-TMP:homebrew-core joe$ brew style bash-snippets
== Formula/bash-snippets.rb ==
C:  9:  1: Extra blank line detected.

1 file inspected, 1 offense detected
```

**Comment by ilovezfs** [comment]
Thanks!

**Comment by alexanderepstein** [comment]
If i use ```assert.equals``` does that check the entire output ? Is there any assert that works only on the first line? The tools i want to test has output that changes (one is weather tool, currency tool etc..) However the first line remains consistent always.

**Comment by ilovezfs** [comment]
You can use `assert_equal` with `shell_output("#{bin}/foo").lines.first` or just `assert_match` to match a substring or regex.

**Comment by alexanderepstein** [comment]
Let me know if those tests are written correctly and work :)

File: Formula/bash-snippets.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  def install
+    system "./install.sh", "--prefix=#{prefix}", "all"
+  end
+
+  test do
+    system "#{bin}/weather", "-h"
```
Comment (ilovezfs) [inline Formula/bash-snippets.rb]: you can leave out each of the `-h` and `-v` tests.

Comment (alexanderepstein) [inline Formula/bash-snippets.rb]: Removed them

**Review by ilovezfs — APPROVED** [review]

**Comment by alexanderepstein** [comment]
Noted. Thank you so much for all your help and working with me to get this into homebrew-core :smile:

**Comment by ilovezfs** [comment]
Thanks for your first contribution to Homebrew, @alexanderepstein! Without people like you submitting PRs we couldn't run this project. You rock!

**Comment by ilovezfs** [comment]
>Thank you so much for all your help and working with me to get this into homebrew-core

You're welcome!

**Pull request merged**